desc/io/input_reader.py

```diff
@@ -776,6 +778,185 @@ def write_desc_input(filename, inputs, header=""):  # noqa: C901 - fxn too compl
+        ), "Equilibrium must have power series profiles for ascii io"
+
+        pres_profile1 = eq0._pressure.params.tolist()
+
+        try:
+            iota_profile1 = eq0._iota.params.tolist()
+            assert (
+                eq0.iota.__class__.__name__ == "PowerSeriesProfile"
+            ), "Equilibrium must have power series profiles for ascii io"
+            char = "i"
+            if len(pres_profile1) >= len(iota_profile1):
+                pres_profile = pres_profile1
+
+                iota_profile = np.zeros((len(pres_profile),))
+                iota_profile[: len(iota_profile1)] = iota_profile1
+            else:
+                pres_profile1 = np.zeros((len(iota_profile1),))
+                pres_profile[: len(pres_profile)] = pres_profile
+        except AttributeError:
+            curr_profile1 = eq0._current.params.tolist()
+            assert (
+                eq0.current.__class__.__name__ == "PowerSeriesProfile"
+            ), "Equilibrium must have power series profiles for ascii io"
+            char = "c"
+            if len(pres_profile1) >= len(curr_profile1):
+                pres_profile = pres_profile1
+
+                curr_profile = np.zeros((len(pres_profile),))
+                curr_profile[: len(curr_profile1)] = curr_profile1
+            else:
+                pres_profile1 = np.zeros((len(curr_profile1),))
+                pres_profile[: len(curr_profile)] = pres_profile
+
+        for l in range(len(pres_profile)):
+            if char == "i":
+                f.write(
+                    "l: {:3d}\tp = {:16.8E}\t{} = {:16.8E}\n".format(
+                        int(l), pres_profile[l], char, iota_profile[l]
+                    )
+                )
+            else:
+                f.write(
+                    "l: {:3d}\tp = {:16.8E}\t{} = {:16.8E}\n".format(
+                        int(l), pres_profile[l], char, curr_profile[l]
+                    )
+                )
+
+        f.write("\n")
+
+        f.write("\n# fixed-boundary surface shape\n")
+        # boundary paramters
+        if eq0.sym:
+            for k, (l, m, n) in enumerate(eq0.surface.R_basis.modes):
+                if abs(eq0.Rb_lmn[k]) > 5e-5:
+                    f.write(
+                        "l: {:3d}\tm: {:3d}\tn: {:3d}\tR1 = {:16.8E}\tZ1 = {:16.8E}\n".format(
+                            int(0), m, n, eq0.Rb_lmn[k], 0
+                        )
+                    )
+            for k, (l, m, n) in enumerate(eq0.surface.Z_basis.modes):
+                if abs(eq0.Zb_lmn[k]) > 5e-5:
+                    f.write(
+                        "l: {:3d}\tm: {:3d}\tn: {:3d}\tR1 = {:16.8E}\tZ1 = {:16.8E}\n".format(
+                            int(0), m, n, 0, eq0.Zb_lmn[k]
+                        )
+                    )
+        else:
+            for k, (l, m, n) in enumerate(eq0.surface.R_basis.modes):
+                if abs(eq0.Rb_lmn[k]) > 5e-5 or abs(eq0.Zb_lmn[k]) > 5e-5:
```

Comment:
5e-5 seems like a pretty high cutoff for ignoring Fourier modes, we should at least make this an argument the user can pass in. Take a look at https://github.com/PlasmaControl/DESC/blob/32398c3ad34e3f05296216126f9091730eb78126/desc/io/input_reader.py#L804

Reply:
I would be surprised if that is the case; Maybe if you want to lower your objective to very small values. I have set the threshold now to 1E-8.